app/src/main/java/lu/geoportail/map/LuxTileServer.kt

```diff
@@ -16,75 +14,115 @@ import java.io.FileOutputStream
+            val file = File(toPathRoot, filename)
+            if (file.exists()) continue
 
-        val sourceFileBytes = resources.openRawResource(
-            context.resources.getIdentifier(resourceName, "raw", packageName)
-        ).readBytes()
-        FileOutputStream(fileName).use {
-            it.write(sourceFileBytes)
+            val sourceFileBytes = assetManager.open("$assetPath/$filename").readBytes()
+            FileOutputStream(file).use {
+                it.write(sourceFileBytes)
+            }
         }
     }
 
-    fun start(filePath: File) {
-        copyResToFile("omt_geoportail_lu", File(filePath,"mbtiles/omt_geoportail_lu.mbtiles"))
-        copyResToFile("omt_topo_geoportail_lu", File(filePath, "mbtiles/omt_topo_geoportail_lu.mbtiles"))
+    fun start() {
+        copyAssets("offline_tiles/mbtiles", File(this.filePath, "mbtiles"))
+//        copyAssets("offline_tiles/styles", File(filePath, "styles"))
+//        copyAssets("offline_tiles/sprites", File(filePath, "sprites"))
 
         val server = AsyncHttpServer()
         server["/", HttpServerRequestCallback { _, response -> response.send("Hello!!!") }]
         server.get("/hello") { request, response -> response.send("Hello!!!$request") }
         server.get("/mbtiles", getMbTile)
         server.get("/static/.*", getStaticFile)
 
-        for ((key, value) in staticsMap) {
-            reverseStaticsMap[value] = key
-        }
         this.db = mapOf(
-            "road" to SQLiteDatabase.openDatabase("${filePath}/mbtiles/omt_geoportail_lu.mbtiles", null, SQLiteDatabase.OPEN_READONLY),
-            "topo" to SQLiteDatabase.openDatabase("${filePath}/mbtiles/omt_topo_geoportail_lu.mbtiles", null, SQLiteDatabase.OPEN_READONLY)
+            "road" to SQLiteDatabase.openDatabase("${this.filePath}/mbtiles/tiles_luxembourg.mbtiles", null, SQLiteDatabase.OPEN_READONLY),
+            "topo" to SQLiteDatabase.openDatabase("${this.filePath}/mbtiles/topo_tiles_luxembourg.mbtiles", null, SQLiteDatabase.OPEN_READONLY)
         )
         // listen on port 8766
         server.listen(8766)
-        // browsing http://localhost:5001 will return Hello!!!
     }
+    private fun replaceUrls(resBytes: ByteArray, resourcePath: String): ByteArray {
+        var resString = String(resBytes)
+        if (resourcePath.contains("_style.json")) {
+            val files = assets.list("offline_tiles/data")
+            // replace in style definition:
+            // mbtiles://{xxx} by
+            //  - http://127.0.0.1:8766/static/data/xxx.json if xxx is found offline
+            //  - https://vectortiles.geoportail.lu/data/xxx.json if xxx is not found offline
+            resString = resString.replace(Regex("mbtiles://\\{(.*)\\}")) {
+                    mm -> mm.groups[1]?.value?.let {
+                        groupValue: CharSequence -> when {
+                            files.any {f -> "$groupValue.json" == f } -> "http://127.0.0.1:8766/static/data/$groupValue.json"
```

Comment:
`files` can be `null` you should check for this.

Reply:
according to type check, it can't be null.
It can be an empty list, but then any is false, so it's all OK